File: source/calculator/plugin_src/ai/rorsch/pandagenie/nativelib/CalculatorLib.java
package ai.rorsch.pandagenie.nativelib;

public class CalculatorLib {

    static {
        System.loadLibrary("calculator");
    }

    public native double add(double a, double b);
    public native double subtract(double a, double b);
    public native double multiply(double a, double b);
    public native double divide(double a, double b);
    public native double modulo(double a, double b);
    public native double power(double base, double exp);
    public native double nSqrt(double x);
    public native double cbrt(double x);
    public native double sin(double x);
    public native double cos(double x);
    public native double tan(double x);
    public native double asin(double x);
    public native double acos(double x);
    public native double atan(double x);
    public native double sinh(double x);
    public native double cosh(double x);
    public native double tanh(double x);
    public native double ln(double x);
    public native double log10(double x);
    public native double log2(double x);
    public native double factorial(int n);
    public native double permutation(int n, int r);
    public native double combination(int n, int r);
    public native double degToRad(double deg);
    public native double radToDeg(double rad);
    public native double evaluate(String expression);
    public native String[] listFunctions();
}

File: source/calculator/jni_bridge/ai/rorsch/pandagenie/nativelib/CalculatorLib.kt
package ai.rorsch.pandagenie.nativelib

class CalculatorLib {

    external fun add(a: Double, b: Double): Double
    external fun subtract(a: Double, b: Double): Double
    external fun multiply(a: Double, b: Double): Double
    external fun divide(a: Double, b: Double): Double
    external fun modulo(a: Double, b: Double): Double
    external fun power(base: Double, exp: Double): Double
    external fun nSqrt(x: Double): Double
    external fun cbrt(x: Double): Double
    external fun sin(x: Double): Double
    external fun cos(x: Double): Double
    external fun tan(x: Double): Double
    external fun asin(x: Double): Double
    external fun acos(x: Double): Double
    external fun atan(x: Double): Double
    external fun sinh(x: Double): Double
    external fun cosh(x: Double): Double
    external fun tanh(x: Double): Double
    external fun ln(x: Double): Double
    external fun log10(x: Double): Double
    external fun log2(x: Double): Double
    external fun factorial(n: Int): Double
    external fun permutation(n: Int, r: Int): Double
    external fun combination(n: Int, r: Int): Double
    external fun degToRad(deg: Double): Double
    external fun radToDeg(rad: Double): Double
    external fun evaluate(expression: String): Double
    external fun listFunctions(): Array<String>
}
